File: source/packages/@ada/common/src/governance/data-product.ts
/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */

/**
 * Defines the different group access permissions for a data product policy
 */
export enum DataProductAccess {
  // Able to query the data product
  READ_ONLY = 'READ_ONLY',
  // Able to query and transform the data product, and also edit the data product (ontology mapping, permissions, etc)
  FULL = 'FULL',
}

File: source/packages/@ada/query-parse-render-lambdas/src/main/java/com/ada/query/parserender/model/rewrite/DataProduct.java
/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */

package com.ada.query.parserender.model.rewrite;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Builder
@Getter
@Setter
@JsonIgnoreProperties(ignoreUnknown = true)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@EqualsAndHashCode
public class DataProduct {
    @SuppressWarnings("squid:S1068")
    private String tableName;

    @SuppressWarnings("squid:S1068")
    private List<DataColumn> columns;
}
